src/Microsoft.Kusto.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -1157,176 +775,15 @@ private bool ShouldSkipIntellisense(string uri)
-                    };
-                },
-                errorHandler: ex =>
-                {
-                    // return error result
-                    return new DefinitionResult
-                    {
-                        IsErrorResult = true,
-                        Message = ex.Message,
-                        Locations = null
-                    };
-                });
-
-            // wait for the queue item
-            queueItem.ItemProcessed.WaitOne();
-            var result = queueItem.GetResultAsT<DefinitionResult>();
-            return result;
-        }
-
-        private DefinitionResult GetDefinitionFromTokenList(TextDocumentPosition textDocumentPosition, List<Token> tokenList,
-                ScriptParseInfo scriptParseInfo, ScriptFile scriptFile, ConnectionInfo connInfo)
-        {
-
-            DefinitionResult lastResult = null;
-            foreach (var token in tokenList)
-            {
-
-                // Strip "[" and "]"(if present) from the token text to enable matching with the suggestions.
-                // The suggestion title does not contain any sql punctuation
-                string tokenText = TextUtilities.RemoveSquareBracketSyntax(token.Text);
-                textDocumentPosition.Position.Line = token.StartLocation.LineNumber;
-                textDocumentPosition.Position.Character = token.StartLocation.ColumnNumber;
-                if (Monitor.TryEnter(scriptParseInfo.BuildingMetadataLock))
-                {
-                    try
-                    {
-                        var result = QueueTask(textDocumentPosition, scriptParseInfo, connInfo, scriptFile, tokenText);
-                        lastResult = result;
-                        if (!result.IsErrorResult)
-                        {
-                            return result;
-                        }
-                    }
-                    catch (Exception ex)
-                    {
-                        // if any exceptions are raised return error result with message
-                        Logger.Write(TraceEventType.Error, "Exception in GetDefinition " + ex.ToString());
-                        return new DefinitionResult
-                        {
-                            IsErrorResult = true,
-                            Message = SR.PeekDefinitionError(ex.Message),
-                            Locations = null
-                        };
-                    }
-                    finally
-                    {
-                        Monitor.Exit(scriptParseInfo.BuildingMetadataLock);
-                    }
-                }
-                else
-                {
-                    Logger.Write(TraceEventType.Error, "Timeout waiting to query metadata from server");
-                }
-            }
-            return (lastResult != null) ? lastResult : null;
-        }
-
         /// <summary>
         /// Get definition for a selected sql object using SMO Scripting
```

Comment:
Please update comment to get definition from DataSource

Reply:
Moved the folder DataSourceIntellisense to DataSOurce folder

Reply:
Changed to "Get definition for a selected text from DataSource."